### ISSUE TITLE
Using imgproc::find_contours_with_hierarchy

### ISSUE THREAD

**Issue by aemino** [twistedfall/opencv-rust#90]
`imgproc::find_contours_with_hierarchy` doesn't seem to work if I pass in a `VectorOfVectorOfint` for the hierarchy parameter. Is a `VectorOfVec4i` type needed?

**Comment by twistedfall** [comment]
Can you please provide the code snippet and the error message you're getting?

**Comment by aemino** [comment]
Code:
```rust
fn find_contours(&self, image: &Mat) -> Vec<VectorOfVectorOfPoint> {
        let mut contours = VectorOfVectorOfPoint::new();
        let mut hierarchy = VectorOfVectorOfint::new();
        imgproc::find_contours_with_hierarchy(
            &image,
            &mut contours,
            &mut hierarchy,
            imgproc::RETR_TREE,
            imgproc::CHAIN_APPROX_SIMPLE,
            opencv::core::Point::new(0, 0),
        )
        .unwrap();

        // ...
}
```

Upon trying to unwrap the `Result` form `find_contours_with_hierarchy`, it errors with the following:
```
thread 'extraction::tests::test_contour_matching' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: -215, message: "OpenCV(4.1.1) /tmp/opencv-20190908-41416-17rm3ol/opencv-4.1.1/modules/core/src/matrix_wrap.cpp:79: error: (-215:Assertion failed) 0 <= i && i < (int)vv.size() in function \'getMat_\'\n" }', src/libcore/result.rs:1165:5
```

Note that this works normally with the non-hierarchical variant.

I've also had success with using a `MatConstIterator` as a workaround, although it isn't particularly ergonomic.

**Comment by twistedfall** [comment]
Judging from the documentation `hierarchy` is supposed to be `vector<Vec4i>`. In the current version this type is not generated, I've just released the fix, so please update to 0.29.3 and use the following code (that works for me):
```rust
let mut contours = opencv::types::VectorOfVectorOfPoint::new();
let mut hierarchy = opencv::types::VectorOfVec4i::new();
imgproc::find_contours_with_hierarchy(
    &image,
    &mut contours,
    &mut hierarchy,
    imgproc::RETR_TREE,
    imgproc::CHAIN_APPROX_SIMPLE,
    opencv::core::Point::new(0, 0),
).unwrap();
```
Generally you can't interchangeably use `VectorOfX` and `VecNX`, these are completely different types. The first one is a bridge to C++ `std::vector`, dynamic and heap-allocated, the latter is OpenCV custom type, fixed-size and stack-allocated akin to Rust's `[X; N]`.

**Comment by aemino** [comment]
Yeah, I'm aware that the VectorOfX and VecNX types have different memory representations, just wanted to try in case there was some conversion mechanism. Thank you for adding the `VectorOfVec4i` type.
